archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/internal/filtering/ArchUnitPropsTestSourceFilter.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    private static boolean isProvided(EngineDiscoveryRequest discoveryRequest, ArchUnitEngineDescriptor engineDescriptor, String key) {
+        return readProperty(discoveryRequest, engineDescriptor, key).isPresent();
+    }
+
+    private static Optional<String> readProperty(EngineDiscoveryRequest discoveryRequest, ArchUnitEngineDescriptor engineDescriptor, String key) {
+        return readFromConfigurationParams(discoveryRequest, key)
+                .or(() -> ArchUnitPropsTestSourceFilter.readFromEngineDescriptor(engineDescriptor, key));
+    }
+
+    private static Optional<String> readFromEngineDescriptor(ArchUnitEngineDescriptor engineDescriptor, String key) {
+        return Optional.ofNullable(engineDescriptor.getConfiguration().getOrDefault(key, null))
+                .map(String.class::cast)
+                .filter(NON_EMPTY_STRING);
+    }
+
+    private static Optional<String> readFromConfigurationParams(EngineDiscoveryRequest discoveryRequest, String key) {
+        return discoveryRequest.getConfigurationParameters().get(toSystemPropertyName(key))
+                .filter(NON_EMPTY_STRING);
+    }
+
+    private static String toSystemPropertyName(String key) {
+        return String.join(".", ARCHUNIT_PREFIX, ArchConfiguration.JUNIT_PREFIX, key);
+    }
+
+    private static class SelectorMatcher {
+
+        private static final String SINGLE_ESCAPED_ASTERISK_STRING = "\\\\*";
+        private static final Pattern MULTIPLE_ESCAPED_ASTERISK = Pattern.compile("(\\\\\\*)+");
+        private static final Pattern SINGLE_ESCAPED_ASTERISK = Pattern.compile("\\*", Pattern.LITERAL);
+
+        private static final Pattern SPECIAL_REGEX_CHARS = Pattern.compile("[{}()\\[\\].+*?^$\\\\|]");
+        private static final SelectorMatcher EMPTY = new SelectorMatcher(new String[0]);
+        private static final SelectorMatcher ACCEPT_ALL = new SelectorMatcher(new String[] {"*"});
+        private static final String ESCAPED_INPUT = "\\\\$0";
+        private static final String MATCH_ALL_CHARACTERS = "\\.\\*";
+
+        private final Collection<Pattern> patterns;
+
+        public SelectorMatcher(String[] patterns) {
+            this.patterns = prepareRegexes(patterns);
+        }
+
+        private Collection<Pattern> prepareRegexes(String[] patterns) {
+            return Arrays.stream(patterns)
+                    .map(this::prepareRegex)
+                    .collect(Collectors.toList());
+        }
+
+        private Pattern prepareRegex(String pattern) {
+            String escaped = SPECIAL_REGEX_CHARS.matcher(pattern).replaceAll(ESCAPED_INPUT);
+            escaped = MULTIPLE_ESCAPED_ASTERISK.matcher(escaped).replaceAll(SINGLE_ESCAPED_ASTERISK_STRING);
+            escaped = SINGLE_ESCAPED_ASTERISK.matcher(escaped).replaceAll(MATCH_ALL_CHARACTERS);
+            return Pattern.compile("(?:^|\\.)" + escaped + "$", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE);
```

Comment:
In case the `:^|\\.` needs explanation, `^` is supposed to match FQ names, while `\\.` is supposed to match simple names (i.e. match the pattern starting from a dot, presumably the final dot in a package name)